m3u_to_playlist.py: Add basic ability to import relative file paths

jicka opened the pull request:
Hello :)
I do not know if you are interested in pull requests. Feel free to ignore if not.

My m3u files contain relative file paths and I run Plex on a dedicated computer, meaning the absolute paths to the music files are different between my computer and on the Plex machine.

I slightly modified the file to be able to deal with this situation. The modification is optional and does not change anything if the file is used in the intended way.

I've added an optional `LibraryPath` argument that provides the script with the relative path to the m3u file within the Plex folder structure. This is then concatenated with the file name in the URL and allows Plex to find the file within its different folder structure. If the argument is not provided, the string stays empty and the concatenation has no effect, keeping the script as it was before.

This only works when the .m3u file sits at the root of the music folder within Plex (and the script is also executed from that location), which are acceptable compromises to avoid heavy modifications of the script, I think.

Warm regards,

Jicka

Casvt commented on playlist_collection/m3u_to_playlist.py:
```diff
@@ -8,6 +8,15 @@
+	- open the terminal from the folder with your music (and where the .m3u file is placed)
+	- run the script with the additional argument "--LibraryPath" including the path of the files within Plex to your music library.
+	Example:
+	linux: python3 ./m3u_to_playlist.py -l "Music" -f "chill.m3u" -u @all -p /media/Music/
```
I would advice to put quotes around the `/media/Music` paths in both examples. As people will take over the examples and fill their own paths in, the example will fall apart when the filepath contains spaces. So just replace it with `... -p "/media/Music"` for both examples.

Casvt commented on playlist_collection/m3u_to_playlist.py:
```diff
@@ -41,7 +50,7 @@ def m3u_to_playlist(ssn, library_name: str, file_path: str, users: list=['@me'])
 		if lib['title'] == library_name:
 			for user_token in user_data.values():
-				ssn.post(f'{base_url}/playlists/upload', params={'sectionID': lib['key'], 'path': file_path, 'X-Plex-Token': user_token})
+				ssn.post(f'{base_url}/playlists/upload', params={'sectionID': lib['key'], 'path': library_path+file_path, 'X-Plex-Token': user_token})
```
Using `+` is a bit dangerous to do. The filepath is for a file on the _server_ so this should be in the _servers os_ format. If you're running Linux but the server is on Windows, then you need to supply a windows path. Using `os.path.join` won't work either as that uses the delimiter of the client, not the server. Best we can do is make it extra clear in the documentation (in argparse and at the top of the file) that the path supplied needs to be in the format of the servers OS and that it should end with that delimiter.

Casvt commented on playlist_collection/m3u_to_playlist.py:
```diff
@@ -58,11 +67,12 @@ def m3u_to_playlist(ssn, library_name: str, file_path: str, users: list=['@me'])
 	# Setup arg parsing
 	parser = ArgumentParser(description='Convert a .m3u file to a plex playlist')
 	parser.add_argument('-l','--LibraryName', type=str, help='Name of target library', required=True)
+	parser.add_argument('-p','--LibraryPath', type=str, help='Path within Plex to the Music files. Make sure it ends with /.', default='', required=False)
```
It doesn't have to end with a `/`. Could also end with a `\` if the server is on Windows...

jicka commented:
Thank you for the detailed feedback. As you see, I'm not very experienced and your commentary was valuable and allowed me to understand the code better.

It turns out, the problem was never relative file paths. Plex just needs a path to find the .m3u by itself (and it seems to deal with relative paths perfectly). So the path provided needs to be correct from Plex's perspective, not the host.

If you're still interested in merging this (feel free to discard it), I've reverted all the previous changes, and switched the `'File not found'` error into a `print()` warning. I've also added a short explanation at the top. See [b8173cd](https://github.com/Casvt/Plex-scripts/pull/208/commits/b8173cd0de6173399f2a300bd890af64538b37ed).

If you're running Plex on your local machine, then the path check is meaningful and will warn you of a potential mistake. If not, then the warning will not prevent the script from running. Sending the wrong path to Plex just has it fail silently, so it's not a problem to continue execution on an erroneous path.

Warm regards,

Jicka
